vms/platformvm/txs/fees/calculator.go

```diff
@@ -62,50 +134,242 @@ func (*Calculator) RewardValidatorTx(*txs.RewardValidatorTx) error {
+	if err != nil {
+		return err
+	}
+
+	_, err = fc.AddFeesFor(complexity)
+	return err
 }
 
-func (fc *Calculator) ExportTx(*txs.ExportTx) error {
-	fc.Fee = fc.config.TxFee
-	return nil
+func (fc *Calculator) ExportTx(tx *txs.ExportTx) error {
+	if !fc.isEActive {
+		fc.Fee = fc.config.TxFee
+		return nil
+	}
+
+	outs := make([]*avax.TransferableOutput, len(tx.Outs)+len(tx.ExportedOutputs))
+	copy(outs, tx.Outs)
+	copy(outs[len(tx.Outs):], tx.ExportedOutputs)
+
+	complexity, err := fc.meterTx(tx, outs, tx.Ins)
+	if err != nil {
+		return err
+	}
+
+	_, err = fc.AddFeesFor(complexity)
+	return err
+}
+
+func (fc *Calculator) meterTx(
+	uTx txs.UnsignedTx,
+	allOuts []*avax.TransferableOutput,
+	allIns []*avax.TransferableInput,
+) (fees.Dimensions, error) {
+	var complexity fees.Dimensions
+
+	uTxSize, err := txs.Codec.Size(txs.CodecVersion, uTx)
+	if err != nil {
+		return complexity, fmt.Errorf("couldn't calculate UnsignedTx marshal length: %w", err)
+	}
+	complexity[fees.Bandwidth] = uint64(uTxSize)
+
+	// meter credentials, one by one. Then account for the extra bytes needed to
+	// serialize a slice of credentials (codec version bytes + slice size bytes)
+	for i, cred := range fc.credentials {
+		c, ok := cred.(*secp256k1fx.Credential)
+		if !ok {
+			return complexity, fmt.Errorf("don't know how to calculate complexity of %T", cred)
+		}
+		credDimensions, err := fees.MeterCredential(txs.Codec, txs.CodecVersion, len(c.Sigs))
+		if err != nil {
+			return complexity, fmt.Errorf("failed adding credential %d: %w", i, err)
+		}
+		complexity, err = fees.Add(complexity, credDimensions)
+		if err != nil {
+			return complexity, fmt.Errorf("failed adding credentials: %w", err)
+		}
+	}
+	complexity[fees.Bandwidth] += wrappers.IntLen // length of the credentials slice
+	complexity[fees.Bandwidth] += codec.VersionSize
+
+	for _, in := range allIns {
+		inputDimensions, err := fees.MeterInput(txs.Codec, txs.CodecVersion, in)
+		if err != nil {
+			return complexity, fmt.Errorf("failed retrieving size of inputs: %w", err)
+		}
+		inputDimensions[fees.Bandwidth] = 0 // inputs bandwidth is already accounted for above, so we zero it
+		complexity, err = fees.Add(complexity, inputDimensions)
+		if err != nil {
+			return complexity, fmt.Errorf("failed adding inputs: %w", err)
+		}
+	}
+
+	for _, out := range allOuts {
+		outputDimensions, err := fees.MeterOutput(txs.Codec, txs.CodecVersion, out)
+		if err != nil {
+			return complexity, fmt.Errorf("failed retrieving size of outputs: %w", err)
+		}
+		outputDimensions[fees.Bandwidth] = 0 // outputs bandwidth is already accounted for above, so we zero it
+		complexity, err = fees.Add(complexity, outputDimensions)
+		if err != nil {
+			return complexity, fmt.Errorf("failed adding outputs: %w", err)
+		}
+	}
+
+	return complexity, nil
+}
+
+func (fc *Calculator) AddFeesFor(complexity fees.Dimensions) (uint64, error) {
+	if fc.feeManager == nil || complexity == fees.Empty {
```

Comment:
this is a bit of defensive programming. `feeMananger` is `nil` for transactions verified pre E upgrade and for these transactions we should not call `AddFeesFor`. However we further add this check to make sure we won't panic over some edge case